### PR TITLE
applications: machine_learning: Add machine learning application

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds machine learning application.

### CODE REVIEW

**Review by pdunaj — APPROVED** [review]

**Review by pdunaj — APPROVED** [review]

File: applications/machine_learning/src/modules/ei_data_forwarder.c [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		LOG_ERR("UART not ready");
+		LOG_ERR("Sampling frequency is too high");
+		report_error();
+		return false;
+	}
+
+	static uint8_t buf[UART_BUF_SIZE];
+	const float *data_ptr = sensor_event_get_data_ptr(event);
+	int pos = 0;
+	int tmp = snprintf(&buf[pos], sizeof(buf) - pos, DATA_FORMAT, *data_ptr);
```
Comment (pdunaj) [inline applications/machine_learning/src/modules/ei_data_forwarder.c:71]: please put that in the look
you will check against data_size==0 plus avoid code duplication

Comment (pdunaj) [reply applications/machine_learning/src/modules/ei_data_forwarder.c:71]: if this is about the comma maybe the following would do
```
for (i = 0; i < 2*size-1; i++) {
   if (i != 0)
      tmp = sprintf(......, ",");
   else
      tmp = sprintf(...., DATA_FORMAT, &data[i/2]);

   error_check(tmp);
}
```

**Review by pdunaj — CHANGES_REQUESTED** [review]

File: applications/machine_learning/src/modules/ei_data_forwarder.c (hunk 1 above)
Comment (MarekPieta) [reply applications/machine_learning/src/modules/ei_data_forwarder.c:71]: Changed

File: applications/machine_learning/src/modules/ei_data_forwarder.c [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	size_t data_cnt = sensor_event_get_data_cnt(event);
+	int pos = 0;
+	int tmp;
+
+	for (size_t i = 0; i < 2 * data_cnt; i++) {
+		if ((i % 2) == 0) {
+			tmp = snprintf(&buf[pos], sizeof(buf) - pos, "%.2f", data_ptr[i / 2]);
+		} else {
+			if (i == (2 * data_cnt - 1)) {
+				tmp = snprintf(&buf[pos], sizeof(buf) - pos, "\r\n");
```
Comment (pdunaj) [inline applications/machine_learning/src/modules/ei_data_forwarder.c:78]: ok but please do
```
if (a) {
} else if (b) {
} else {
}
```

**Review by pdunaj — CHANGES_REQUESTED** [review]

File: applications/machine_learning/src/modules/ei_data_forwarder.c (hunk 2 above)
Comment (MarekPieta) [reply applications/machine_learning/src/modules/ei_data_forwarder.c:78]: Changed

**Review by pdunaj — APPROVED** [review]

**Comment by MarekPieta** [comment]
The changes will be merged in a separate PRs. I close this one.

**Pull request closed without merging**